### DISCUSSION TITLE
Dropwizard Server Restarts automatically

### DISCUSSION THREAD

**Discussion by AhangarAamir** [dropwizard/dropwizard#9010] [General]
Why sometimes dropwizard server restarts automatically can someone help me here?

**Comment by joschi** [comment]
@AhangarAamir Please provide more information about how your Dropwizard applications are being deployed.

Are there any messages in the logs of the Dropwizard applications or the runtime environment when the restart happens?

**Reply by AhangarAamir** [reply]
@joschi i am running Dropwizard in a kubernetes pod having 4GB of memory, below are the logs 
```
d.a.r.UserActionResource -  listing user actions for orgId : dcc3ad51-ee3f-4165-a979-24af6dfe1aa1 and spaceId : 5612bf75-aa19-4c3a-a61c-36b38ac8eee7 and actorId : null 
INFO   [11:18:13.921] [dw-18 - GET /dstore/org/5612bf75-aa19-4c3a-a61c-36b38ac8eee7/stats?orgId=dcc3ad51-ee3f-4165-a979-24af6dfe1aa1&expandable=false&userId=] c.d.d.a.r.ProjectResource -  Found 2 datasets 
INFO   [11:18:13.930] [dw-18 - GET /dstore/org/5612bf75-aa19-4c3a-a61c-36b38ac8eee7/stats?orgId=dcc3ad51-ee3f-4165-a979-24af6dfe1aa1&expandable=false&userId=] c.d.d.a.r.ProjectResource -  Found 1 models 
INFO   [11:18:14.059] [dw-20 - GET /dstore/org/5612bf75-aa19-4c3a-a61c-36b38ac8eee7/stats?userId=bb0661c8-4fc2-4625-ab08-465bf05d0c84&orgId=dcc3ad51-ee3f-4165-a979-24af6dfe1aa1] c.d.d.a.r.ProjectResource -  Found 2 datasets 
INFO   [11:18:14.069] [dw-20 - GET /dstore/org/5612bf75-aa19-4c3a-a61c-36b38ac8eee7/stats?userId=bb0661c8-4fc2-4625-ab08-465bf05d0c84&orgId=dcc3ad51-ee3f-4165-a979-24af6dfe1aa1] c.d.d.a.r.ProjectResource -  Found 1 models 
INFO   [11:19:39.996] [main] o.e.j.u.log -  Logging initialized @8390ms to org.eclipse.jetty.util.log.Slf4jLog 
INFO   [11:19:40.489] [main] i.d.s.DefaultServerFactory -  Registering jersey handler with root path prefix: / 
INFO   [11:19:40.494] [main] i.d.s.DefaultServerFactory -  Registering admin handler with root path prefix: / 
INFO   [11:19:40.494] [main] i.d.a.AssetsBundle -  Registering AssetBundle with name: io.paradoxical.dropwizard.swagger.bundles.SwaggerUIBundle for path /dropwizard-swagger/* 
```
You can check the logs; Dropwizard registers the components again. During this registration process, my service endpoints return a 503 error

**Reply by joschi** [reply]
This is the normal startup. What happens before and which events happen for your Kubernetes pods during that time?

https://kubernetes.io/docs/reference/kubectl/generated/kubectl_events/

**Reply by AhangarAamir** [reply]
@joschi Sir, this is not a normal startup. Before registering Dropwizard components, my service was running. You can check that some GET requests are happening. After that, components started registering again, and the Kubernetes pod was running with zero restarts.

**Reply by joschi** [reply]
Please post the complete logs of your application, preferably with DEBUG level.
